test/jdk/javax/swing/JSplitPane/TestSplitPaneResetDividerLoc.java

```diff
@@ -55,64 +58,82 @@ public class TestSplitPaneResetDividerLoc {
     private static JDialog dialog;
     private static JFrame frame;
     private static volatile Point point;
+    private static volatile Rectangle size;
     private static volatile int setLoc;
     private static volatile int curLoc;
 
-    public static void main(String[] args) throws Exception {
+     private static void setLookAndFeel(UIManager.LookAndFeelInfo laf) {
         try {
-            SwingUtilities.invokeAndWait(TestSplitPaneResetDividerLoc::createAndShowUI);
-
-            Robot robot = new Robot();
-            robot.waitForIdle();
-            robot.delay(1000);
-            SwingUtilities.invokeAndWait(() -> {
-                point = openDialogButton.getLocationOnScreen();
-            });
-            robot.mouseMove(point.x, point.y);
-            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
-            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
-            robot.waitForIdle();
-            robot.delay(1000);
-
-            SwingUtilities.invokeAndWait(() -> {
-                int divLoc = splitPane.getDividerLocation();
-                splitPane.setDividerLocation(divLoc + 200);
-            });
-            robot.waitForIdle();
-            robot.delay(1000);
-            robot.keyPress(KeyEvent.VK_ESCAPE);
-            robot.keyRelease(KeyEvent.VK_ESCAPE);
-
-            SwingUtilities.invokeAndWait(() -> {
-                setLoc = splitPane.getDividerLocation();
-                System.out.println(setLoc);
-            });
-
-            robot.waitForIdle();
-            robot.delay(1000);
-            robot.mouseMove(point.x, point.y);
-            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
-            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
-            robot.waitForIdle();
-            robot.delay(1000);
-
-            SwingUtilities.invokeAndWait(() -> {
-                curLoc = splitPane.getDividerLocation();
-                System.out.println(curLoc);
-            });
-
-            robot.keyPress(KeyEvent.VK_ESCAPE);
-            robot.keyRelease(KeyEvent.VK_ESCAPE);
-
-            if (curLoc != setLoc) {
-                throw new RuntimeException("Divider location is preserved");
-            }
-        } finally {
-            SwingUtilities.invokeAndWait(() -> {
-                if (frame != null) {
-                    frame.dispose();
+            UIManager.setLookAndFeel(laf.getClassName());
+        } catch (UnsupportedLookAndFeelException ignored) {
+            System.out.println("Unsupported LAF: " + laf.getClassName());
+        } catch (ClassNotFoundException | InstantiationException
+                 | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        for (UIManager.LookAndFeelInfo laf : UIManager.getInstalledLookAndFeels()) {
+            System.out.println("Testing LAF : " + laf.getClassName());
+            try {
+                setLookAndFeel(laf);
```

Comment:
If setLookAndFeel encounters UnsupportedLookAndFeelException we will output the exception into stdout but we will do the entire test again with the previously installed LAF. Shall we do so? If we consider that exception as non-fatal for the test maybe we can make setLookAndFeel  method boolean instead of void returning true only if it was successfully installed new LAF and skip test iteration if it returns false?